### PR TITLE
fix: update supportedMessageTypes structure

### PR DESCRIPTION
**Pull request by ujjwal-ab**
`#` Description

- Resolved INT-131
- This PR accommodates handling of the new `supportedMessageTypes` structure.
- eg. ```
"supportedMessageTypes": {
			"cloud": ["track", "group"],
			"device":{
					"web": ["track", "group"],
					"android": ["track", "group"]
				}
                          }
```


## Linear Ticket

[INT-131](https://linear.app/rudderstack/issue/INT-131/review-the-configuration-attributes-required-in-configuration-files)

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by sanpj2292 — APPROVED** [review]

File: processor/eventfilter/eventfilter.go [hunk 1]
```diff
@@ -24,20 +24,22 @@ var pkgLogger = logger.NewLogger().Child("eventfilter")
...
-				if !skip {
-					supportedMessageTypes = append(supportedMessageTypes, supportedType)
+		if supportedTypesMap, ok := supportedTypes.(map[string]interface{}); ok {
+			if supportedTypeInterface, ok := supportedTypesMap["cloud"].([]interface{}); ok {
+				supportedTypesArr := misc.ConvertInterfaceToStringArray(supportedTypeInterface)
+				for _, supportedType := range supportedTypesArr {
+					var skip bool
+					switch supportedType {
+					case "identify":
+						skip = identifyDisabled(destination)
```
Comment (koladilip) [inline processor/eventfilter/eventfilter.go:34]: when we have supportMessageTypes when again why need again identifyDisabled (enableServerSideIdentify)?

**Comment by Sidddddarth** [comment]
LGTM.
My only question is how is the rollout for this change planned, considering it expects a change in the config response from control plane.

File: processor/eventfilter/eventfilter.go [hunk 2]
```diff
@@ -24,20 +24,22 @@ var pkgLogger = logger.NewLogger().Child("eventfilter")
...
-			for _, supportedType := range supportedTypesArr {
-				var skip bool
-				switch supportedType {
-				case "identify":
-					skip = identifyDisabled(destination)
-				default:
-				}
-				if !skip {
-					supportedMessageTypes = append(supportedMessageTypes, supportedType)
+		if supportedTypesMap, ok := supportedTypes.(map[string]interface{}); ok {
```
Comment (atzoum) [inline processor/eventfilter/eventfilter.go:27]: When did this schema change happened in control plane?

**Comment by sanpj2292** [comment]
@ujjwal-ab let's figure out a way to make sure we have backward compatibility wrt the changes mentioned in this PR

**Comment by ujjwal-ab** [comment]
Closing, not required

**Pull request closed without merging**